### PR TITLE
test: disable tcp_connect6_link_local on android

### PR DESCRIPTION
**Pull request by bnoordhuis**
The test fails with ENETUNREACH on CI, presumably because the Android image doesn't support link-local traffic.

### CODE REVIEW

**Review by cjihrig — APPROVED** [review]

**Comment by bnoordhuis** [comment]
Maybe (hopefully) #4490 invalidates this PR so I'll wait until that one is merged, then rebase this one.

**Review by saghul — APPROVED** [review]

**Pull request closed without merging**
